Analyzing RNAseq for differential expression
-------------------------------------------

```{r}
library(edgeR)
```

###read in raw count data per gene

```{r}
counts <- read.delim("../requisiteData/sam2countsResults.tsv",row.names=1)
boxplot(counts)

#Look at boxplot of log2
normexplog <- log(counts,2) 
boxplot(normexplog)
```

```{r}
#check the file
head(counts)
```

```{r, results='hide'}
summary(counts)
```

Library wtbother1.4 is disturbingly small, I am going to remove it from analysis. 

```{r}
counts <- subset(counts, select=-c(wtbother1.4))
```

Need to convert NA to 0 counts
```{r}
counts[is.na(counts)] <- 0
```

We are adding a new column "group", that identifies which library belongs to which group sample type.

```{r}
group <- c(
  rep("tf2ambr", 4), 
  rep("tf2aother", 4), 
  rep("tf2bmbr", 3), 
  rep("tf2bother", 5), 
  rep("tf2cmbr", 4),
  rep("tf2cother", 4),
  rep("wtambr", 3),
  rep("wtaother", 4),
  rep("wtbmbr", 4),
  rep("wtbother", 3),
  rep("wtcmbr", 6),
  rep("wtcother", 3)
)
```


Now we have to put the data into a DGEList object. [Here is a nice summary](http://davetang.org/muse/2012/01/19/the-dgelist-object-in-r/) of what the DGEList object is all about.

```{r}
d <- DGEList(counts=counts, group=group)
```

Check to make sure each group matches the correct sample.  

```{r}
head(d$counts)
d$samples


#Filter to exclude genes that have <2 counts in (N Rep)-1
cpm.d<- cpm(d)
d <- d[rowSums(cpm.d>1)>=3,] 
             
#normalize library
d <- calcNormFactors(d)
d$pseudo<- equalizeLibSizes(d)
write.csv (d$pseudo$pseudo,"normalized_read_count.csv")

# read in norm data
data_norm <- read.csv("normalized_read_count.csv", row.names=1)
dim(data_norm) 
```

MDS using dist and cmdscale
==========================

library
```{r}
library(ggplot2)
```
Prepare for MDS plot
```{r}
m <- cmdscale(dist(scale(t(data_norm))))
colnames(m) <- c("x", "y")
m<-as.data.frame(m)
```

Label for visualization

```{r}
m$sampleSection <- c(
  rep("ambr", 4), 
  rep("aother", 4), 
  rep("bmbr", 3), 
  rep("bother", 5), 
  rep("cmbr", 4),
  rep("cother", 4),
  rep("ambr", 3),
  rep("aother", 4),
  rep("bmbr", 4),
  rep("bother", 3),
  rep("cmbr", 6),
  rep("cother", 3)
)

m$genotype <- c(
  rep("tf2", 24),
  rep("wt", 23)
)

m$type <- c(
  rep("mbr", 4), 
  rep("other", 4), 
  rep("mbr", 3), 
  rep("other", 5), 
  rep("mbr", 4),
  rep("other", 4),
  rep("mbr", 3),
  rep("other", 4),
  rep("mbr", 4),
  rep("other", 3),
  rep("mbr", 6),
  rep("other", 3)
)
```

###All the samples 
There is no visible trends, even when subsetting by sample. 

```{r}
ggplot(m, aes(x,y,colour=sampleSection))+geom_point(size=5)+theme_bw() 
ggplot(m, aes(x,y,colour=sampleSection))+geom_point(size=5)+theme_bw() + facet_grid(.~ genotype)
```

###By Genotype
There is no visible trend
```{r}
ggplot(m, aes(x,y,colour=genotype))+geom_point(size=5)+theme_bw()
```

###All together 
The only time a slight trend appears is between Marginal Blastozone Regions (MBR) and rachis regions (other).

```{r}
ggplot(m, aes(x,y,colour=type, label=sampleSection))+geom_point(size=1)+theme_bw() + geom_text() 
ggplot(m, aes(x,y,colour=type, label=sampleSection))+geom_point(size=1)+theme_bw() + geom_text() + facet_grid(. ~ genotype)
```

library(rmarkdown)
render("MDS.Rmd", "pdf_document")
