sanalyzing RNAseq for differential expression of LCM data
---------------------------------------------------------
script modified from a script given to me by Aashish Ranjan called `edgeR_DE.R`

Ciera Martinez

##Install

`source("http://bioconductor.org/biocLite.R")`
`biocLite("edgeR")`

```{r}
library(edgeR)
```

##Read in Data
Read in raw count data per gene.
Add checknames to FALSE because it was making the columns unique. 

```{r, results = 'hide'}
counts <- read.delim("../sam2countsResults.tsv",row.names=1)

#check the file
head(counts)
summary(counts)
colnames(counts)
#need to convert NA to 0 counts
counts[is.na(counts)] <- 0
```

Subset per DE expirement
========================
I am going to start by subsetting the particular treatments I am looking at.

####WT
###Marginal Blastozone vs Other

```{r}
wtcregion <- counts[,40:48]
head(wtcregion)
```

```{r}
#convert data to a form that edgeR wants
group <- c(rep("wtcmbr", 6), rep("wtcother",3))
d <- DGEList(counts=wtcregion,group=group)
d$samples 
```

Computes counts per million (CPM) then,
Filter to exclude genes that have <2 counts in (N Rep)-1
```{r}
cpm.d<- cpm(d)
d <- d[rowSums(cpm.d>2)>=3,] 
```

Estimate Common Negative Binomial Dispersion by Conditional Maximum Likelihood. Maximizes the negative binomial conditional common likelihood to give the estimate of the common dispersion across all tags.

```{r}
d <- estimateCommonDisp(d,verbose=T)
```

Normalize library
```{r}
d <- calcNormFactors(d)
```
     
Estimate overdispersion
Important so that the correct model is fit

```{r}
d <- estimateCommonDisp(d)
```

Calculate DE genes

```{r}
DEtest <- exactTest(d,pair=c("wtcmbr","wtcother"))
head(DEtest$table)
```

Create a table of the results, with multiple testing correction.  

```{r}
results <- topTags(DEtest) #removed #n=Inf not sure what that means ask Aashish.
head(results$table) 
```

These are the topTags, but I want to continue with all the DE genes.  Right?
How many genes are DE?


```{r}
dim(DEtest$table)
sum(DEtest$table$PValue<.05)  #changed from sum(results$table$adj.P.Val<.01) which resulted in 0
```

#How many genes in each direction?
```{r}
summary(decideTestsDGE(DEtest,p.value=.05))
```

Plot the results
First create a table of DE to highlight those with p < 0.01
```{r}
sig.genes <- rownames(DEtest$table[DEtest$table$PValue<0.05,])
#sig.genes <- rownames(results$table[results$table$adj.P.Val<0.01,]) #original which returned 0 characters
```

Visualize with smear plot
```{r}
plotSmear(d,de.tags=sig.genes)
```

###Subset by significant score
```{r}
results.sig <- subset(DEtest$table, DEtest$table$PValue < 0.01)
```
What are the genes that are misexpressed?
For this we need to add some annotation

```{r, results = 'head'}
annotation1<- read.delim("../ITAG2.3_all_Arabidopsis_ITAG_annotations.tsv", header=FALSE)  #Changed to the SGN human readable annotation
colnames(annotation1)<- c("ITAG", "SGN_annotation")
annotation2<- read.delim ("../ITAG2.3_all_Arabidopsis_annotated.tsv")
annotation <- merge (annotation1,annotation2, by =1,1, all.x=TRUE)
head(annotation)
#head(results) This returns "Error: Two subscripts required"

DEtest.annotated <- merge(results.sig,annotation,by.x="row.names",by.y="ITAG",all.x=T,sort=F)

#was 
#results.annotated <- merge(results$table,annotation,by.x="row.names",by.y="ITAG",all.x=T,sort=F)

head(DEtest.annotated,n=30)
```

#Write table with results
write.table(DEtest.annotated,"wtcmbr_wtcother_DE.txt",sep="\t",row.names=F)



