sanalyzing RNAseq for differential expression of LCM data
---------------------------------------------------------
script modified from a script given to me by Aashish Ranjan called `edgeR_DE.R`

Ciera Martinez

##Install

`source("http://bioconductor.org/biocLite.R")`
`biocLite("edgeR")`

```{r}
library(edgeR)
```

##Read in Data
Read in raw count data per gene.
Add checknames to FALSE because it was making the columns unique. 

```{r, results = 'hide'}
counts <- read.delim("../sam2countsResults.tsv",row.names=1)

#check the file
head(counts)
summary(counts)
colnames(counts)
#need to convert NA to 0 counts
counts[is.na(counts)] <- 0
```

Subset per DE expirement
========================
I am going to start by subsetting the particular treatments I am looking at.

####WT
###Marginal Blastozone vs Other

```{r}
wtcregion <- counts[,40:48]
head(wtcregion)
```

```{r}
#convert data to a form that edgeR wants
group <- c(rep("wtcmbr", 6), rep("wtcother",3))
d <- DGEList(counts=wtcregion,group=group)
d$samples 
```

Computes counts per million (CPM) then,
Filter to exclude genes that have <2 counts in (N Rep)-1
```{r}
cpm.d<- cpm(d)
d <- d[rowSums(cpm.d>2)>=3,] 
```

Estimate Common Negative Binomial Dispersion by Conditional Maximum Likelihood. Maximizes the negative binomial conditional common likelihood to give the estimate of the common dispersion across all tags.

```{r}
d <- estimateCommonDisp(d,verbose=T)
```

Normalize library
```{r}
d <- calcNormFactors(d)
```
     
Estimate overdispersion
Important so that the correct model is fit

```{r}
d <- estimateCommonDisp(d)
```

Calculate DE genes

```{r}
DEtest <- exactTest(d,pair=c("wtcmbr","wtcother"))
head(DEtest$table)
```

Create a table of the results, with multiple testing correction.  

```{r}
results <- topTags(DEtest, n=Inf)
head(results)
```

These are the topTags, but I want to continue with all the DE genes. 
How many genes are DE?

#How many genes in each direction?
```{r}

dim(results$table)
sum(results$table$FDR<.05) # How many are DE genes?
summary(decideTestsDGE(DEtest,p.value=.05))

```

Plot the results
First create a table of DE to highlight those with p < 0.05

```{r}
sig.genes <- rownames(results$table[results$table$FDR<0.05,]) 
```

Visualize with smear plot
```{r}
plotSmear(d,de.tags=sig.genes)
```

```{r}
dim(results$table)
sum(results$table$FDR<.05) # How many are DE genes?
summary(decideTestsDGE(DEtest,p.value=.05))

sig.genes <- rownames(results$table[results$table$FDR<0.05,]) 
plotSmear(d,de.tags=sig.genes)
```

What are the genes that are misexpressed?
For this we need to add some annotation


Essentially we are merging two annotations files to 1.) only sig genes 2.) all genes
```{r}

annotation1<- read.delim("../ITAG2.3_all_Arabidopsis_ITAG_annotations.tsv", header=FALSE)  #Changed to the SGN human readable annotation
colnames(annotation1) <- c("ITAG", "SGN_annotation")
annotation2<- read.delim ("../ITAG2.3_all_Arabidopsis_annotated.tsv")
annotation <- merge(annotation1,annotation2, by = "ITAG")

#Making the only significant gene table
results.sig$ITAG <- rownames(results.sig)  #change row.names to ITAG for merging
results.sig.annotated <- merge(results.sig,annotation,by = "ITAG") #This is merging to only sig genes

#Making all table

results$table$ITAG <- rownames(results$table)
results.all.annotated <- merge(results$table, annotation,by = "ITAG") #This s
```


Write table with results

```{r}
write.table(results.all.annotated,"wtcmbr_wtcother_DE_all.txt",sep="\t",row.names=F)
write.table(results.sig.annotated,"wtcmbr_wtcother_DE.txt",sep="\t",row.names=F)
```