#Exploring Differential Expression Venn Diagrams 

This script takes all the differential expressed (DE) regions between the "rachis" and MBR and looks at what are the similarities in (DE) genes between these region along the longitudinal axis.  

DE was previously performed on each region (A, B, and C) between both types of tissue (Marginal Blastozone Region (MBR) and "rachis").  So the question becomes: Are there shared genes between all the up-regulated genes in each tissue type along the longitudinal axis of the leaf?

To answer this question, I took each the DE gene from each region and 1.) visulized the the overlapp between each region (A, B, and C) with a venn diagram. and 2.) Wrote out which genes belong to each of the venn diagrams in their subsequent folders, splitting the analysis up into three categories: all, up-regulated, and down regulated.  The up-regulated genes are those genes up-regulated in the rachis compared to the MBR regions, while the down-regulated are those that are up-regulated in the MBR compared to rachis. 

##Script

To run script and create a report at the end, run `rmarkdown` code below.

```{r, eval=FALSE}
library(rmarkdown)
render("vennDiagram.tf2.Rmd", "pdf_document")
```


Read in required libraries.

```{r}
library(VennDiagram)
```

##Upload annotation data for table outputs

```{r}
annotation1<- read.delim("../requisiteData/ITAG2.3_all_Arabidopsis_ITAG_annotations.tsv", header=FALSE)  #Changed to the SGN human readable annotation
colnames(annotation1) <- c("ITAG", "SGN_annotation")
annotation2<- read.delim ("../requisiteData/ITAG2.3_all_Arabidopsis_annotated.tsv")
annotation <- merge(annotation1,annotation2, by = "ITAG")
```


##Rachis Categories

Between region along the longitudinal axis, (A, B, C), differential gene expression was performed between the "rachis" regions and the Marginal Blastozone Regions.

First read in data and force logFC to numeric.
```{r}
tf2A <- read.table("../2014.6.22_analysis.tf2/tf2ambr_tf2aother/tf2ambr_tf2aother_DE_sig.txt", header = TRUE, fill = TRUE) #read table
tf2A$logFC <- as.numeric(as.character(tf2A$logFC)) #force numeric

tf2B <- read.table("../2014.6.22_analysis.tf2/tf2bmbr_tf2bother/tf2bmbr_tf2bother_DE_sig.txt", header = TRUE, fill = TRUE)
tf2B$logFC <- as.numeric(as.character(tf2B$logFC))

tf2C <- read.table("../2014.6.22_analysis.tf2/tf2cmbr_tf2cother/tf2cmbr_tf2cother_DE_sig.txt", header = TRUE, fill = TRUE)
tf2C$logFC <- as.numeric(as.character(tf2C$logFC))
```

Subset genes that are significantly Up regulated between the "rachis" region and the "other" region. Then subset genes that are significantly Down regulated between the "rachis" region and the "other" regions.

```{r}
tf2Aup <- subset(tf2A, logFC > 0, select = c(ITAG, logFC, SGN_annotation, gene_name, AGI))
tf2Bup <- subset(tf2B, logFC > 0, select = c(ITAG, logFC, SGN_annotation, gene_name, AGI))
tf2Cup <- subset(tf2C, logFC > 0, select = c(ITAG, logFC, SGN_annotation, gene_name, AGI))

tf2Adown <- subset(tf2A, logFC < 0, select = c(ITAG, logFC, SGN_annotation, gene_name, AGI))
tf2Bdown <- subset(tf2B, logFC < 0, select = c(ITAG, logFC, SGN_annotation, gene_name, AGI))
tf2Cdown <- subset(tf2C, logFC < 0, select = c(ITAG, logFC, SGN_annotation, gene_name, AGI))
```

##ALL GENES
This takes all significant gene regardless of up-regulated or down-regulated.

```{r}
dir.create("all") #Creates directory to put all out tables.

#To get unique


#ABall
ABall <- matrix(intersect(tf2A$ITAG, tf2B$ITAG))
colnames(ABall) <- c("ITAG")
ABall.annotated <- merge(ABall, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(ABall.annotated, file = "all/ABall.annotated.csv") #write .csv file

#BCall
BCall <- matrix(intersect(tf2B$ITAG, tf2C$ITAG))
colnames(BCall) <- c("ITAG") #name the first column for calling the column 
BCall.annotated <- merge(BCall, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(BCall.annotated, file = "all/BCall.annotated.csv") #write .csv file

#ACall
ACall <- matrix(intersect(tf2A$ITAG, tf2C$ITAG))
colnames(ACall) <- c("ITAG")
ACall.annotated <- merge(ACall, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(ACall.annotated, file = "all/ACall.annotated.csv") #write .csv file

#ABCall
ABCall <- matrix(intersect(ABall, tf2C$ITAG))
colnames(ABCall) <- c("ITAG")
ABCall.annotated <- merge(ABCall, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(ABCall.annotated, file = "all/ABCall.annotated.csv") #write .csv file

#Makes the grid
grid.newpage()
venn.plot <- draw.triple.venn(area1    = length(tf2A$ITAG),
                              area2    = length(tf2B$ITAG),
                              area3    = length(tf2C$ITAG),
                              n12      = length(ABall),
                              n23      = length(BCall),
                              n13      = length(ACall),
                              n123     = length(ABCall),
                              category = c("A", "B", "C"),
                              cat.pos  = c(0, 40, 250),
                              cat.dist = c(0.05, 0.05, 0.05),
                              fill     = c("blue", "red", "green"),
                              alpha    = 0.3,
                              lty      = "blank",
                              cex      = 2,
                              cat.cex  = 2,
                              cat.col  = c("blue", "red", "green"))
grid.draw(venn.plot)

#Get only genes specific to A
onlyAall <- matrix(setdiff(tf2A$ITAG, tf2B$ITAG))
onlyAall <- matrix(setdiff(onlyAall, tf2C$ITAG))
colnames(onlyAall) <- c("ITAG")
onlyAall.annotated <- merge(onlyAall, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(onlyAall.annotated, file = "all/onlyAall.annotated.csv") #write .csv file

#Get only genes specific to B
onlyBall <- matrix(setdiff(tf2B$ITAG, tf2A$ITAG))
onlyBall <- matrix(setdiff(onlyBall, tf2C$ITAG))
colnames(onlyBall) <- c("ITAG")
onlyBall.annotated <- merge(onlyBall, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(onlyBall.annotated, file = "all/onlyBall.annotated.csv") #write .csv file

#Get only genes specific to B
onlyCall <- matrix(setdiff(tf2C$ITAG, tf2A$ITAG))
onlyCall <- matrix(setdiff(onlyCall, tf2B$ITAG))
colnames(onlyCall) <- c("ITAG")
onlyCall.annotated <- merge(onlyCall, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(onlyCall.annotated, file = "all/onlyCall.annotated.csv") #write .csv file
```


##UP Regulation Genes
Up regulated means significatly upregulated in the "rachis"" region compared to the MBR.

```{r}
dir.create("up") #Creates directory to put all out tables.

#ABup
ABup <- matrix(intersect(tf2Aup$ITAG, tf2Bup$ITAG))
colnames(ABup) <- c("ITAG")
ABup.annotated <- merge(ABup, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(ABup.annotated, file = "up/ABup.annotated.csv") #write .csv file

#BCup
BCup <- matrix(intersect(tf2Bup$ITAG, tf2Cup$ITAG))
colnames(BCup) <- c("ITAG")
BCup.annotated <- merge(BCup, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(BCup.annotated, file = "up/BCup.annotated.csv") #write .csv file

#ACup
ACup <- matrix(intersect(tf2Aup$ITAG, tf2Cup$ITAG))
colnames(ACup) <- c("ITAG")
ACup.annotated <- merge(ACup, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(ACup.annotated, file = "up/ACup.annotated.csv") #write .csv file

#ABCup
ABCup <- matrix(intersect(ABup, tf2Cup$ITAG))
colnames(ABCup) <- c("ITAG")
ABCup.annotated <- merge(ABCup, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(ABCup.annotated, file = "up/ABCup.annotated.csv") #write .csv file
dim(ABCup)

#Make Plot

grid.newpage()
venn.plot <- draw.triple.venn(area1    = length(tf2Aup$ITAG),
                              area2    = length(tf2Bup$ITAG),
                              area3    = length(tf2Cup$ITAG),
                              n12      = length(ABup),
                              n23      = length(BCup),
                              n13      = length(ACup),
                              n123     = length(ABCup),
                              category = c("A", "B", "C"),
                              cat.pos  = c(0, 40, 250),
                              cat.dist = c(0.05, 0.05, 0.05),
                              fill     = c("blue", "red", "green"),
                              alpha    = 0.3,
                              lty      = "blank",
                              cex      = 2,
                              cat.cex  = 2,
                              cat.col  = c("blue", "red", "green"))
grid.draw(venn.plot)

#Get only genes specific to A
onlyAup <- matrix(setdiff(tf2Aup$ITAG, tf2Bup$ITAG))
onlyAup <- matrix(setdiff(onlyAup, tf2Cup$ITAG))
colnames(onlyAup) <- c("ITAG")
onlyAup.annotated <- merge(onlyAup, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(onlyAup.annotated, file = "up/onlyAup.annotated.csv") #write .csv file

#Get only genes specific to B
onlyBup <- matrix(setdiff(tf2Bup$ITAG, tf2Aup$ITAG))
onlyBup <- matrix(setdiff(onlyBup, tf2Cup$ITAG))
colnames(onlyBup) <- c("ITAG")
onlyBup.annotated <- merge(onlyBup, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(onlyBup.annotated, file = "up/onlyBup.annotated.csv") #write .csv file

#Get only genes specific to B
onlyCup <- matrix(setdiff(tf2Cup$ITAG, tf2Aup$ITAG))
onlyCup <- matrix(setdiff(onlyCup, tf2Bup$ITAG))
colnames(onlyCup) <- c("ITAG")
onlyCup.annotated <- merge(onlyCup, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(onlyCup.annotated, file = "up/onlyCup.annotated.csv") #write .csv file
```

##Down Regulated Genes

Down regulated genes means significantly down regulated in "rachis" compared to other.  Or you can think of it as significanly up regulated in MBR compared with rachis.

```{r}

dir.create("down") #Creates directory to put all out tables.

#ABdown
ABdown <- matrix(intersect(tf2Adown$ITAG, tf2Bdown$ITAG))
colnames(ABdown) <- c("ITAG")
ABdown.annotated <- merge(ABdown, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(ABdown.annotated, file = "down/ABdown.annotated.csv") #write .csv file

#BCdown
BCdown <- matrix(intersect(tf2Bdown$ITAG, tf2Cdown$ITAG))
colnames(BCdown) <- c("ITAG")
BCdown.annotated <- merge(BCdown, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(BCdown.annotated, file = "down/BCdown.annotated.csv") #write .csv file

#ACdown
ACdown <- matrix(intersect(tf2Adown$ITAG, tf2Cdown$ITAG))
colnames(ACdown) <- c("ITAG")
ACdown.annotated <- merge(ACdown, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(ACdown.annotated, file = "down/ACdown.annotated.csv") #write .csv file

#ABCdown
ABCdown <- matrix(intersect(ABdown, tf2Cdown$ITAG))
colnames(ABCdown) <- c("ITAG")
ABCdown.annotated <- merge(ABCdown, annotation,by = "ITAG", by.x = TRUE) #Merge with annotated
write.csv(ABCdown.annotated, file = "down/ABCdown.annotated.csv") #write .csv file


#Make Plot
grid.newpage()
venn.plot <- draw.triple.venn(area1    = length(tf2Adown$ITAG),
                              area2    = length(tf2Bdown$ITAG),
                              area3    = length(tf2Cdown$ITAG),
                              n12      = length(ABdown),
                              n23      = length(BCdown),
                              n13      = length(ACdown),
                              n123     = length(ABCdown),
                              category = c("A", "B", "C"),
                              cat.pos  = c(0, 40, 250),
                              cat.dist = c(0.05, 0.05, 0.05),
                              fill     = c("blue", "red", "green"),
                              alpha    = 0.3,
                              lty      = "blank",
                              cex      = 2,
                              cat.cex  = 2,
                              cat.col  = c("blue", "red", "green"))
grid.draw(venn.plot)

#Get only genes specific to A
onlyAdown <- matrix(setdiff(tf2Adown$ITAG, tf2Bdown$ITAG))
onlyAdown <- matrix(setdiff(onlyAdown, tf2Cdown$ITAG))
colnames(onlyAdown) <- c("ITAG")
onlyAdown.annotated <- merge(onlyAdown, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(onlyAdown.annotated, file = "down/onlyAdown.annotated.csv") #write .csv file

#Get only genes specific to B
onlyBdown <- matrix(setdiff(tf2Bdown$ITAG, tf2Adown$ITAG))
onlyBdown <- matrix(setdiff(onlyBdown, tf2Cdown$ITAG))
colnames(onlyBdown) <- c("ITAG")
onlyBdown.annotated <- merge(onlyBdown, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(onlyBdown.annotated, file = "down/onlyBdown.annotated.csv") #write .csv file

#Get only genes specific to B
onlyCdown <- matrix(setdiff(tf2Cdown$ITAG, tf2Adown$ITAG))
onlyCdown <- matrix(setdiff(onlyCdown, tf2Bdown$ITAG))
colnames(onlyCdown) <- c("ITAG")
onlyCdown.annotated <- merge(onlyCdown, annotation,by = "ITAG", by.x = TRUE, all.x = TRUE) #Merge with annotated
write.csv(onlyCdown.annotated, file = "down/onlyCdown.annotated.csv") #write .csv file
```




