# Question 1

1. **Are there genes that are differentially expressed between MBR and Rachis in all three groups across the longitudinal axis?** Genotype specific. Basically asking what are the genes that define the Marginal Blastozone. What are the genes that define the Rachis. I am most interested in what is occuring in WT. This can be performed in each genotype and compared. **Hypothesis**: There will be genes that overlap between the analysis done between the two genotypes because the Marginal Blastozone.  

## Key to Samples

genotype: either wildtype of *tf2*

region: A. tip B. early emmerging leaflet C. base

type: MBR = Marginal Blastozone Region, other = the rachis or midvein region

### libraries
```{r}
library(edgeR)
```

###Read in Data
Read in raw count data per gene.

```{r, results='hide'}
counts <- read.delim("../../data/sam2countsResults.tsv")

colnames(counts)

#Remove tf2ambr.3, tf2aother7 and tf2aother8 because read count is so low.  See readsReps.Rmd for full report.

counts <- counts[,-2]

#check the file
head(counts)
colnames(counts)

#need to convert NA to 0 counts
counts[is.na(counts)] <- 0
```

## Design matrix 

```{r}
#make the groups
head(counts)
colnames(counts)

sample <- gsub("[0-9]", "", names(counts))
sample <- gsub("\\.", "", sample)
sample <- sample[-1]
sample

#set genotype
designTable <- as.data.frame(sample)

designTable$genotype <- ifelse(grepl("wt", designTable$sample, ignore.case = T), "wt", ifelse(grepl("tf", designTable$sample, ignore.case = T), "tf2", "unknown"))

#set type

designTable$tissue <- ifelse(grepl("other", designTable$sample, ignore.case = T), "rachis", ifelse(grepl("mbr", designTable$sample, ignore.case = T), "mbr", "unknown"))

#Set Region
designTable$region <- ifelse(grepl("a", designTable$sample, ignore.case = T), "A", ifelse(grepl("c", designTable$sample, ignore.case = T), "C", "B"))

designTable
genotype <- designTable$genotype 
sample <- designTable$sample
tissue <- designTable$region
region <- designTable$r
```

```{r}
#put into DGE List
y <- DGEList(counts=counts[,2:48], genes=counts[,1], group = sample)
head(y)

cpm.y <- cpm(y) #counts per million
y <- y[rowSums(cpm.y > 5) >= 3,] # get rid of genes with low counts

y <- estimateCommonDisp(y,verbose=T) #Estimates common negative binomial dispersion by conditional maximum liklihood

y <- calcNormFactors(y)

y <- estimateCommonDisp(y, verbose = T)

y$samples

plotMDS(cpm(y, log=TRUE), column=1)

