# Question 1

1. **Are there genes that are differentially expressed between MBR and Rachis in all three groups across the longitudinal axis?** Genotype specific. Basically asking what are the genes that define the Marginal Blastozone. What are the genes that define the Rachis. I am most interested in what is occuring in WT. This can be performed in each genotype and compared. **Hypothesis**: There will be genes that overlap between the analysis done between the two genotypes because the Marginal Blastozone.  

## Key to Samples

genotype: either wildtype of *tf2*

region: A. tip B. early emmerging leaflet C. base

type: MBR = Marginal Blastozone Region, other = the rachis or midvein region

### libraries
```{r}
library(edgeR)
library(locfit)
library(statmod)
```

###Read in Data
Read in raw count data per gene.

```{r, results='hide'}
counts <- read.delim("../../data/sam2countsResults.tsv")

colnames(counts)

#Remove tf2ambr.3, wtcmbr.1.4.6 and tf2aother7 because read count is so low.  See readsReps.Rmd for full report.

counts <- counts[,-c(3,9,42)]

#check the file
head(counts)
colnames(counts)

#need to convert NA to 0 counts
counts[is.na(counts)] <- 0
```

## Normalization 

```{r}
#make the groups

head(counts)
colnames(counts)

sample <- gsub("[0-9]", "", names(counts))
sample <- gsub("\\.", "", sample)
sample <- sample[-1]
sample

#set genotype
designTable <- as.data.frame(sample)

designTable$genotype <- ifelse(grepl("wt", designTable$sample, ignore.case = T), "wt", ifelse(grepl("tf", designTable$sample, ignore.case = T), "tf2", "unknown"))

#set type

designTable$tissue <- ifelse(grepl("other", designTable$sample, ignore.case = T), "rachis", ifelse(grepl("mbr", designTable$sample, ignore.case = T), "mbr", "unknown"))

#Set Region
designTable$region <- ifelse(grepl("a", designTable$sample, ignore.case = T), "A", ifelse(grepl("c", designTable$sample, ignore.case = T), "C", "B"))

designTable
genotype <- designTable$genotype 
sample <- designTable$sample
tissue <- designTable$region
region <- designTable$r
```

```{r}
#put into DGE List

y <- DGEList(counts=counts[,2:46], genes=counts[,1], group = sample)
head(y)

cpm.y <- cpm(y) #counts per million
y <- y[rowSums(cpm.y > 5) >= 3,] # get rid of genes with low counts

y <- estimateCommonDisp(y,verbose=T) #Estimates common negative binomial dispersion by conditional maximum liklihood

y <- calcNormFactors(y)
y <- estimateCommonDisp(y, verbose = T) #Disp = 0.46228 , BCV = 0.6799 
y$samples

plotMDS(cpm(y, log=TRUE), column=1) #Disp = 0.44804 , BCV = 0.6694 
```

There are a few outliers, but these are kept in as they may reflect how homogeneous these tissues are. They are only in the 
tf2 mutant. 

## Analysis - WT

In order to answer the question: Are there genes that are differentially expressed between MBR and Rachis tissue in all three groups across the longitudinal axis? I am going to use an additive linear model, with the longitudinal axis as the blocking factor. This should be genotype specific.


```{r}
#First I need to subset based on genotype

colnames(y)
wtY<- y[,22:45]

designTableWT <- designTable[22:45,]

designTableWT
wtRegion <- designTableWT$region
wtTissue <- designTableWT$tissue

design <- model.matrix(~wtRegion + wtTissue)

rownames(design) <- colnames(wtY)

design

wtY <- estimateDisp(wtY, design, robust=TRUE) #Estimate Dispersion

wtY$common.dispersion # 0.3489676

plotBCV(wtY)
```

### WT- Differential Gene Expression

```{r}
fit <- glmFit(wtY, design)

lrt <- glmLRT(fit)

topTags(lrt,200)
```

Here we see the top tags for MBR vs rachis tissue differential. It adjusted for baseline differences between top, middle, and base. 

```{r}
o <- order(lrt$table$PValue)

cpm(wtY)[o[1:10],]

```

```{r}
detags <- rownames(y)[as.logical(de)]
plotSmear(lrt, de.tags=detags)
abline(h=c(-1, 1), col="blue")
```

All gene output
```{r}
results <- topTags(lrt, n=Inf)

dim(results$table)
summary(de <- decideTestsDGE(lrt))

sum(results$table$FDR<.05) # How many are DE genes total?

#Subset only significant
results.sig <- subset(results$table, results$table$FDR < 0.05) 

sig.genes <- results.sig$genes #only gene names
```

What are the genes that are misexpressed?
For this we need to add some annotation.

Essentially we are merging two annotations files to 1.) only sig genes 2.) all genes

```{r}
annotation1<- read.delim("../../data/ITAG2.3_all_Arabidopsis_ITAG_annotations.tsv", header=FALSE)  #Changed to the SGN human readable annotation
colnames(annotation1) <- c("ITAG", "SGN_annotation")
annotation2<- read.delim("../../data/ITAG2.3_all_Arabidopsis_annotated.tsv")
annotation <- merge(annotation1,annotation2, by = "ITAG")
head(annotation)

#Making the only significant gene table
colnames(results.sig)[1] <- "ITAG" #change to ITAG for merging with annotation

colnames(results.sig)[1]<- "itag"
colnames(annotation)[1] <- "itag"

results.sig.annotated <- merge(results.sig, annotation, by = "itag", all.x=TRUE) #This is merging only sig genes

colnames(results.sig.annotated )
head(results.sig.annotated)

#Making all table

results$table$ITAG <- rownames(results$table)
results.all.annotated <- merge(results$table, annotation,by = "ITAG")

#Write out table to file
write.table(results.all.annotated, file = "WT.allresults.question1.txt", sep = "\t",row.names=F) 

write.table(results.sig.annotated, file = "WT.onlysigresults.question1.txt", sep = "\t",row.names=F) 
```

## Set-Up
```{r}
sessionInfo()
```
